src/batch/src/task/task_manager.rs

```diff
@@ -212,7 +216,27 @@ impl BatchManager {
     /// Called by global memory manager for total usage of batch tasks. This variable should be
     /// maintained by Batch Manager.
     pub fn get_all_memory_usage(&self) -> usize {
-        todo!()
+        let mut all = 0;
+        let mut max_mem_task_id = None;
+        let mut max_mem = usize::MIN;
+        let guard = self.tasks.lock();
```

Comment:
The `get_all_memory_usage` method should be called quite frequently, so using a lock here may be inappropriate. It's better that we update the `all_memory_usage` field in every report, and this function only returns the value. 
Finding the max mem task id should be called only when necessary, e.g. we need to kill some task. That should not be as frequent as this one, and using a lock when necessary is appropriate.

Reply:
> It's better that we update the all_memory_usage field in every report, and this function only returns the value.

Yes, I have considered this. But if we only store it as one value -- all `report` will add bytes on this value, how to avoid the duplicate computation on same task? For example, each task may report the memory bytes but BatchManager do not know how and when to set to 0.

Therefore, single value seems not enough, there has to be some Map to record all task's memory. A lock seems hard to avoid..

Reply:
> Finding the max mem task id should be called only when necessary, e.g. we need to kill some task. That should not be as frequent as this one, and using a lock when necessary is appropriate.

This is good, so it maybe reasonable to find out the highest task only when need to request killing tasks to save memory.

Reply:
> Yes, I have considered this. But if we only store it as one value -- all report will add bytes on this value, how to avoid the duplicate computation on same task? For example, each task may report the memory bytes but BatchManager do not know how and when to set to 0.

For this, we don't need to call the set 0 method, we just need to remember last reported mem value of each task, and tell batch task context the diff, and then apply the diff to total value. It reminds me that storing an atomic counter in `BatchManager` may affect performance, and we should use hytra.

Reply:
Let me clarify this, the correct way should be:

The report thread (`allocation_stat`) should tell TaskContext every time the bytes value, and TaskContext should store the last one and current one. Based on the diff, the report thread then directly update the `all_memory_usage` (single value) in BatchManager with that diff. 

This looks feasible. 🤔

Reply:
Updated